### PR TITLE
fixed docker for dev environment to have vendors synchronized

### PR DESCRIPTION
**Pull request by AlessandroMinoccheri**
In this PR I fixed the docker environment for development to have vendors synchronized out of the container.
In this way, it's easier to add/remove dependencies.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -43,11 +43,7 @@ services:
         php-fpm
       "
     volumes:
-      - ./config:/application/config:rw
-      - ./src:/application/src
-      - ./public:/application/public
-      - ./templates:/application/templates
-      - ./tests:/application/tests
+      - .:/application
```
Comment (JellyBellyDev) [inline docker-compose.yml:46]: this is the solution that I had proposed a few months ago that @fabiocicerchia rejected me. I err?

File: sys/docker/alpine-phpfpm/Dockerfile [hunk 2]
```diff
@@ -53,6 +53,8 @@ ENV LD_PRELOAD /usr/lib/preloadable_libiconv.so php
 
 COPY --from=composer:2 /usr/bin/composer /usr/local/bin/composer
 
+RUN composer self-update
```
Comment (JellyBellyDev) [inline sys/docker/alpine-phpfpm/Dockerfile:56]: sure?

Comment (AlessandroMinoccheri) [reply sys/docker/alpine-phpfpm/Dockerfile:56]: Installing dependencies from zero returns a errors, we need to keep composer updated.

Comment (JellyBellyDev) [reply sys/docker/alpine-phpfpm/Dockerfile:56]: also after update composer.lock anche symfony.lock?

Comment (JellyBellyDev) [reply sys/docker/alpine-phpfpm/Dockerfile:56]: and so, if we commit composer.lock and symfony.lock updated, we can remove this line?

Comment (AlessandroMinoccheri) [reply sys/docker/alpine-phpfpm/Dockerfile:56]: I don't think so, we should install composer 2.1 into our docker image if we would like to remove this line.

Comment (JellyBellyDev) [reply sys/docker/alpine-phpfpm/Dockerfile:56]: `COPY --from=composer:2.1 /usr/bin/composer /usr/local/bin/composer`
don't work?

Comment (AlessandroMinoccheri) [reply sys/docker/alpine-phpfpm/Dockerfile:56]: I can try, it could be better.

Comment (garak) [reply sys/docker/alpine-phpfpm/Dockerfile:56]: Maybe we could force the minor version of composer? Last one is 2.3

Comment (AlessandroMinoccheri) [reply sys/docker/alpine-phpfpm/Dockerfile:56]: Without `composer self-update` I had problems with dependencies. So we can try to remove the line and commit `composer.lock` and `symfony.lock`.

Comment (fabiocicerchia) [reply sys/docker/alpine-phpfpm/Dockerfile:56]: I would add those files to git :+1:

Comment (AlessandroMinoccheri) [reply sys/docker/alpine-phpfpm/Dockerfile:56]: Those 2 files are already committed

Comment (AlessandroMinoccheri) [reply sys/docker/alpine-phpfpm/Dockerfile]: I removed the `composer self-update` at the moment. I verified and now it seems to work fine.

**Review by JellyBellyDev — APPROVED** [review]

**Pull request merged**